io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -456,8 +456,9 @@ pub trait LvsLvol: LogicalVolume + Share {
     /// Returns a boolean indicating if the lvol is a snapshot.
     fn is_snapshot(&self) -> bool;
 
-    /// Returns a boolean indicating if the lvol is a clone.
-    fn is_clone(&self) -> bool;
+    /// Lvol is considered as clone if its sourceuuid attribute is a valid
+    /// snapshot. if it is clone, return the snapshot lvol.
+    fn is_snapshot_clone(&self) -> (bool, Option<Lvol>);
```

Comment:
Why not simply return Option<Lvol> ?

Reply:
Yes, it can be done with single Option parameter as return type. Changed the code.